### ISSUE TITLE
Ability to convert absolute counters into incremental counters

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#5763]
A Vector user has requested the ability to convert absolute counters into incremental counters. I believe this is due to the fact that the downstream system requires incremental counters. Before starting work on this we should answer the following questions:

1. Is the above assumption correct? If it is a requirement of a downstream storage, then this conversion should happen automatically in the sink. (is this part of the new metric buffer changes?)
2. What other conversions make sense and how would we generalize that?

**Comment by bruceg** [comment]
I think this needs to happen in either the relevant sink or in the metrics batch buffer, since the sinks involved only support one or the other. See https://github.com/timberio/vector/issues/5589#issuecomment-750381902 for more discussion.

**Comment by binarylogic** [comment]
Agree with @bruce. I'm closing this in favor of that. Each sink should be handling this conversion as necessary. See #422 for an example.
